### ISSUE TITLE
False positive in rule `closure_spacing`

### ISSUE THREAD

**Issue by Jeehut** [realm/SwiftLint#4272]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

In the following code, the rule `closure_spacing` is giving me a warning where I would not expect to see one:

```swift
itemEntry.reduce(into: [String: String]()) { dict, pair in dict[String(pair.0)] = pair.1 }
```

### Environment

Version 0.49.1
Installed vie Homebrew

**Comment by marcelofabri** [comment]
Are you sure this snippet by itself triggers the violation? I couldn't repro it in `main` or with `0.49.1`. Any chance this is inside another closure in the actual code?

**Comment by Jeehut** [comment]
@marcelofabri Here's the full wrapper, I hope it's reproducible with this:

```swift
import Foundation

extension XmlParsing {
   enum AndroidStrings {
      struct PluralsEntry: Parser {
         func parse(_ input: inout Substring) throws -> AndroidStringsEntry {
            let (attributes, quantityTranslations): (OrderedDictionary<String, String>, [String: String]) = try Parse {
               Many {
                  // ...
               }
               .map { itemEntry in
                  itemEntry.reduce(into: [String: String]()) { dict, pair in dict[String(pair.0)] = pair.1 }  // <-- here's the warnings
               }
            }
            .parse(&input)

            // ...
         }
      }
   }
}

```

**Comment by jpsim** [comment]
I can repro with your example on `main`, but it's a legitimate violation because of the trailing comment.

If I remove `  // <-- here's the warnings` comment then there's no violation.

**Comment by Jeehut** [comment]
Well, it occurred to me without the comment, I just added that here on GitHub. I can't share the full code as is, sorry if that's not helpful. Also I don't understand how the comment makes it a legitimate violation, what does a comment have to do with "closure spacing"? Anyhow, I don't currently have the time to investigate further, feel free to close this if it's really not reproducible and no one else has the same issue.
